models/db/engine_test.go

```diff
@@ -84,3 +88,37 @@ func TestPrimaryKeys(t *testing.T) {
 		}
 	}
 }
+
+func TestSlowQuery(t *testing.T) {
```

Comment:
The test is not right.

I know you like long discussions like https://codeberg.org/forgejo/discussions/issues/54 (for testing), but I think you should first understand the code correctly and write tests correctly.

Reply:
I do agree with @wxiaoguang, this test looks really weird, and not like it tests what it is supposed to test.
I don't even know what the intention of this test is, the implementation looks straightforward enough for me not to warrant a test.
There are far worse things without tests that should have a test.

Reply:
Removed.